api/server/volume.go

```diff
@@ -774,12 +774,20 @@ func (vd *volAPI) enumerate(w http.ResponseWriter, r *http.Request) {
 	for i, s := range ids {
 		vol, err := volumes.Inspect(ctx, &api.SdkVolumeInspectRequest{VolumeId: s})
 		if err != nil {
-			e := fmt.Errorf("Failed to inspect volumeID: %s", err.Error())
-			vd.sendError(vd.name, method, w, e.Error(), http.StatusBadRequest)
+			if s, ok := status.FromError(err); ok && s.Code() != codes.NotFound {
```

Comment:
i think the logic here is a little confusing because if it is ever not `ok` then it will fall through to line 783.

I think you want `ok && s.Code() == codes.NotFound` then move lines 784-785 to inside the if, the move 778-779 out of the if scope